test/integration/standard_filter_test.rb

```diff
@@ -861,7 +861,7 @@ def test_all_filters_never_raise_non_liquid_exception
       { foo: "bar" },
       [{ "foo" => "bar" }, { "foo" => 123 }, { "foo" => nil }, { "foo" => true }, { "foo" => ["foo", "bar"] }],
       { 1 => "bar" },
-      ["foo", 123, nil, true, false, Drop, ["foo"], { foo: "bar" }],
```

Comment:
I think passing a `Class` here is something simply not supported as it cannot be done in a context that runs outside the standard filters. 

Before the current changes, you will see a `e.respond_to?(:to_liquid) ? e.to_liquid : e` in `lib/liquid/standardfilters.rb`. 

Only the StandardFilters apply that `respond_to?` and I believe it is the user responsibility to always pass in a "Liquid compatible" object back to the context and not for Liquid to silence the unexpected input.